Feature/api design 1.0

SamRagusa opened the pull request:
One thing that should probably still be added is setting attributes from a pandas dataframe.

StokesMIDE committed:
Start of skeleton/stubs of cloud connection class. Incomplete.
Docstrings need completion, type hinting required

StokesMIDE committed:
Fleshing out API skeleton

StokesMIDE committed:
Added some of Steve's example code (test)

StokesMIDE committed:
Added `account_email` and `account_id`

SamRagusa committed:
Added Unfinished Methods 

- Added the code for the unfinished methods and functions (for the cloud api)

StokesMIDE reviewed (changes requested):
The only real request for a change is the bit about using `self.domain`; the rest are mostly comments and suggestions.

SamRagusa commented on endaq/cloud/core.py:
```diff
@@ -0,0 +1,296 @@
+
+        devices = {}
+        for f_data in files_json_data:
+            if f_data['device'] is not None and len(f_data['device']) > 0:
```
This doesn't work, if `f_data['device']` is None, then what you're suggesting will evaluate to True, but I need it to evaluate to False in that case.   (I've also tested it and it fails in that case)

CrepeGoat replied:
@StokesMIDE correct me if I'm wrong, but I think this is what you're getting at:
```pycon
>>> def v1(obj):
...     return (obj is not None and len(obj) > 0)
...
>>> def v2(obj):
...     return bool(obj)
...
>>> o = None
>>> assert v1(o) == v2(o)
>>> o = []
>>> assert v1(o) == v2(o)
>>> o = ["non-empty list"]
>>> assert v1(o) == v2(o)
```

In that case, I think you mean `if f_data['device']`

SamRagusa committed:
Made Changes Requested From PR Comments

- Added some more context to potential errors
- improved use of the endaq 'environment' (like dev, production, stagin)
- Added a few comments

SamRagusa committed:
Remove The utilities.py File 

- Removed the utilities.py file since it's contents have moved to core.py

StokesMIDE commented on endaq/cloud/core.py:
```diff
@@ -0,0 +1,307 @@
+"""
+Core enDAQ Cloud communication API
+
+TODO: Rewrite docstrings (pref. before implementing methods)
```
Just delete this line and I think it's good to go.

StokesMIDE reviewed (changes requested):
One minor change and all's good.

SamRagusa committed:
Removed A TODO
